example-crates/example_electrum/src/main.rs

```diff
@@ -296,7 +278,9 @@ fn main() -> anyhow::Result<()> {
 
         let indexed_additions = {
             let mut additions = IndexedAdditions::<ConfirmationHeightAnchor, _>::default();
-            let (_, index_additions) = graph.index.reveal_to_target_multi(&final_update.keychain);
+            let (_, index_additions) = graph
+                .index
+                .reveal_to_target_multi(&final_update.last_active_indices);
```

Comment:
This is not specifically to do with this PR. I think we are making suboptimal use of our API. Only in `scan` do we need to do this work so it should only occur in that branch to make that clear to the user. The changes can be staged there and only committed out here.

Reply:
With sync, it's still possible to find a transaction that will increase last active indices right?

Reply:
No.

Reply:
In sync you don't need to "look ahead" you are only scanning the addresses you've already revealed.